client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java

```diff
@@ -356,32 +228,19 @@ private Exception instantiateUnexpectedException(SwaggerMethodParser methodParse
         Exception result;
         try {
             final Constructor<? extends RestException> exceptionConstructor = exceptionType.getConstructor(String.class, HttpResponse.class, exceptionBodyType);
-
-            boolean isSerializableContentType = contentType == null || contentType.isEmpty()
-                    || contentType.startsWith("application/json")
-                    || contentType.startsWith("text/json")
-                    || contentType.startsWith("application/xml")
-                    || contentType.startsWith("text/xml");
-
-            final Object exceptionBody = responseContent.isEmpty() || !isSerializableContentType
-                    ? null
-                    : serializer.deserialize(responseContent, exceptionBodyType, bodyEncoding(response.headers()));
-
-            result = exceptionConstructor.newInstance("Status code " + responseStatusCode + ", " + bodyRepresentation, response, exceptionBody);
-        } catch (IllegalAccessException | InstantiationException | InvocationTargetException | NoSuchMethodException | JsonParseException e) {
+            result = exceptionConstructor.newInstance("Status code " + responseStatusCode + ", " + bodyRepresentation, response, response.deserializedBody());
```

Comment:
Slight behavior change: it's no longer so easy to distinguish whether the error body was absent or it failed to deserialize. I'd have to save the JsonParsingException somewhere when decoding. So now I'm returning RestException if the error body failed to deserialize, but at least the raw content will still be visible.